src/Query/QueryResult.php

```diff
@@ -501,18 +502,14 @@ public function toArray() {
 	 *
 	 * @return integer|null
 	 */
-	public function getTotalCount() {
-		$query = $this->mQuery;
-		$query->querymode = \SMWQuery::MODE_COUNT; //change query format to count
-		$store = $this->getStore();
-		try {
-			$test = $store->getQueryResult( $query );
-		} catch (\Exception|\Throwable $e) {
-			return null;
-		}
-		$queryRes = $store->getQueryResult( $query );
-		$total = $queryRes instanceof \SMWQueryResult ? $queryRes->getCountValue() : null;
-		return $total;
+	public function getTotalCount(): int {
+		$store = $this->mStore;
+		$queryStr = self::getQueryString();
+		$processedParams = \SMWQueryProcessor::getProcessedParams( [] );
+		$countQuery = \SMWQueryProcessor::createQuery( $queryStr, $processedParams, \SMWQueryProcessor::INLINE_QUERY, 'count' );
+		$queryRes = $store->getQueryResult( $countQuery );
+		$total = $queryRes instanceof \SMWQueryResult ? $queryRes->getCountValue() : 0;
+		return $total ? $total : 0;
```

Comment:
return $total ?? 0;

That is shorter and makes it clear the check is for `null`